{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.1",
    "cveMetadata": {
        "cveId": "CVE-2024-8088",
        "assignerOrgId": "28c92f92-d60d-412d-b760-e73465c3df22",
        "state": "PUBLISHED",
        "assignerShortName": "PSF",
        "dateReserved": "2024-08-22T12:42:32.661Z",
        "datePublished": "2024-08-22T18:45:31.807Z",
        "dateUpdated": "2024-08-23T00:09:14.061Z"
    },
    "containers": {
        "cna": {
            "affected": [
                {
                    "defaultStatus": "unaffected",
                    "product": "CPython",
                    "repo": "https://github.com/python/cpython",
                    "vendor": "Python Software Foundation",
                    "versions": [
                        {
                            "lessThanOrEqual": "3.13.0",
                            "status": "affected",
                            "version": "0",
                            "versionType": "python"
                        }
                    ]
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "type": "remediation developer",
                    "value": "Jason R. Coombs"
                },
                {
                    "lang": "en",
                    "type": "coordinator",
                    "value": "Seth Larson"
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "supportingMedia": [
                        {
                            "base64": false,
                            "type": "text/html",
                            "value": "<div>There is a HIGH severity vulnerability affecting the CPython \"zipfile\"\nmodule.<br></div><div><br></div><div>When iterating over names of entries in a zip archive (for example, methods\nof \"zipfile.ZipFile\" like \"namelist()\", \"iterdir()\", \"extractall()\", etc)\nthe process can be put into an infinite loop with a maliciously crafted\nzip archive. This defect applies when reading only metadata or extracting\nthe contents of the zip archive. Programs that are not handling\nuser-controlled zip archives are not affected.</div>"
                        }
                    ],
                    "value": "There is a HIGH severity vulnerability affecting the CPython \"zipfile\"\nmodule.\n\n\n\n\n\nWhen iterating over names of entries in a zip archive (for example, methods\nof \"zipfile.ZipFile\" like \"namelist()\", \"iterdir()\", \"extractall()\", etc)\nthe process can be put into an infinite loop with a maliciously crafted\nzip archive. This defect applies when reading only metadata or extracting\nthe contents of the zip archive. Programs that are not handling\nuser-controlled zip archives are not affected."
                }
            ],
            "metrics": [
                {
                    "cvssV4_0": {
                        "Automatable": "NO",
                        "Recovery": "USER",
                        "Safety": "NEGLIGIBLE",
                        "attackComplexity": "LOW",
                        "attackRequirements": "NONE",
                        "attackVector": "NETWORK",
                        "baseScore": 8.7,
                        "baseSeverity": "HIGH",
                        "privilegesRequired": "NONE",
                        "providerUrgency": "NOT_DEFINED",
                        "subAvailabilityImpact": "NONE",
                        "subConfidentialityImpact": "NONE",
                        "subIntegrityImpact": "NONE",
                        "userInteraction": "NONE",
                        "valueDensity": "NOT_DEFINED",
                        "vectorString": "CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/S:N/AU:N/R:U/RE:L",
                        "version": "4.0",
                        "vulnAvailabilityImpact": "HIGH",
                        "vulnConfidentialityImpact": "NONE",
                        "vulnIntegrityImpact": "NONE",
                        "vulnerabilityResponseEffort": "LOW"
                    },
                    "format": "CVSS",
                    "scenarios": [
                        {
                            "lang": "en",
                            "value": "GENERAL"
                        }
                    ]
                },
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    },
                    "format": "CVSS",
                    "scenarios": [
                        {
                            "lang": "en",
                            "value": "GENERAL"
                        }
                    ]
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-835",
                            "description": "CWE-835 Loop with Unreachable Exit Condition ('Infinite Loop')",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "28c92f92-d60d-412d-b760-e73465c3df22",
                "shortName": "PSF",
                "dateUpdated": "2024-08-22T18:56:11.589Z"
            },
            "references": [
                {
                    "tags": [
                        "vendor-advisory"
                    ],
                    "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/GNFCKVI4TCATKQLALJ5SN4L4CSPSMILU/"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/pull/122906"
                },
                {
                    "tags": [
                        "issue-tracking"
                    ],
                    "url": "https://github.com/python/cpython/issues/122905"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/795f2597a4be988e2bb19b69ff9958e981cb894e"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/8c7348939d8a3ecd79d630075f6be1b0c5b41f64"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/dcc5182f27c1500006a1ef78e10613bb45788dea"
                }
            ],
            "source": {
                "discovery": "UNKNOWN"
            },
            "title": "Infinite loop when iterating over zip archive entry names",
            "x_generator": {
                "engine": "Vulnogram 0.2.0"
            }
        },
        "adp": [
            {
                "title": "CVE Program Container",
                "references": [
                    {
                        "url": "http://www.openwall.com/lists/oss-security/2024/08/22/1"
                    },
                    {
                        "url": "http://www.openwall.com/lists/oss-security/2024/08/22/4"
                    }
                ],
                "providerMetadata": {
                    "orgId": "af854a3a-2127-422b-91ae-364da2661108",
                    "shortName": "CVE",
                    "dateUpdated": "2024-08-23T00:09:14.061Z"
                }
            }
        ]
    }
}